### PR TITLE
Add SMTPConfig directive for Net::SMTP options & SMTP AUTH support

### PR DESCRIPTION
**Pull request by jonjensen**
Add configuration to catalog.cfg and/or interchange.cfg like this:

SendMailProgram Net::SMTP

SMTPConfig <<END
    host smtp.gmail.com
    user you@your.domain
    pass ucogeygtbhbzgayu
    ssl 1
    timeout 5
END

Also available: port (Net::SMTP defaults to 25 when ssl is off, 465 when
ssl is on)

Note that when using user & pass (SMTP AUTH), Net::SMTP requires Authen::SASL.

### CODE REVIEW

**Comment by jonjensen** [comment]
@msjohns1 Would you mind reviewing this for me? When done, please don't merge it yet. I'm waiting for some more client feedback first and might as well get that before merging. Thanks!

File: lib/Vend/Util.pm
Comment (msjohns1) [inline lib/Vend/Util.pm:2245]: Where does logError() go during startup? Does it end up at stdout the way logGlobal() does? This seems like it would be good information to see in the terminal during startup, along frankly with the contents of $@.

**Review by msjohns1 — APPROVED** [review]
All looks good. Would like to be sure error on `require Net::SMTP` shows in the output stream on a restart, and include the actual error.

File: lib/Vend/Util.pm
Comment (jonjensen) [reply lib/Vend/Util.pm]: Yeah, I think at startup all the logs go to stderr.

It would be good to check this at startup, but I'm not willing to rearchitect this whole thing now to do that. I was just adding logging here for the otherwise opaque way Net::SMTP failures were historically logged, but I don't want to change *when* the error is raised as part of this SMTP AUTH feature.

You're totally right that `$@` should be included in the message, so I just pushed a revised commit that does that.

**Pull request merged**
